src/otaclient/_utils.py

```diff
@@ -41,26 +41,53 @@ def wait_and_log(
     check_interval: int = 2,
     log_interval: int = 30,
     log_func: Callable[[str], None] = logger.info,
-) -> None:
-    """Wait for <flag> until it is set while print a log every <log_interval>."""
+    timeout: int | None = None,
+) -> bool:
+    """Wait for <flag> until it is set while print a log every <log_interval>.
+
+    Args:
+        check_flag: Function that returns a boolean value to check.
+        message: Message to include in the log output.
+        check_for: The value to check against (True or False).
+        check_interval: How often to check the flag in seconds.
+        log_interval: How often to log a message in seconds.
+        log_func: Function to use for logging.
+        timeout: Maximum time to wait in seconds. None means wait indefinitely.
+
+    Returns:
+        bool: True if condition was met, False if timeout occurred.
+    """
     log_round = 0
     for seconds in itertools.count(step=check_interval):
         if check_flag() == check_for:
-            return
+            return True
+
+        # Check if timeout has been reached
+        if timeout and seconds >= timeout:
+            log_func(f"Timeout reached after {timeout}s waiting for {message}")
+            return False
 
         _new_log_round = seconds // log_interval
         if _new_log_round > log_round:
             log_func(f"wait for {message}: {seconds}s passed ...")
             log_round = _new_log_round
         time.sleep(check_interval)
+    return False
 
 
-def check_other_otaclient(pid_fpath: StrOrPath) -> None:  # pragma: no cover
+def check_other_otaclient(
+    pid_fpath: StrOrPath, is_skip: bool
+) -> None:  # pragma: no cover
     """Check if there is another otaclient instance running, and then
     create a pid lock file for this otaclient instance.
 
     NOTE that otaclient should not run inside a PID namespace.
     """
+    # Skip check if the condition is met
+    if is_skip:
+        logger.info("Skipping otaclient multiple instances check")
+        return
+
```

Comment:
if a specific environment variable is set, skip to check multiple otaclient running.